### ISSUE TITLE
WELD_VERTICES flag enabled leads to invalid triangle meshes

### ISSUE THREAD

**Issue by jpx** [IfcOpenShell/IfcOpenShell#4]
If `WELD_VERTICES` is disabled resulting scene looks fine.
When `WELD_VERTICES` is enabled, part of the triangles are totally lost, on several ifc files, some triangles seem to degenerate.

The issue is reproducible with test asset
`IfcOpenShell\test\input\acad2010_walls.ifc`

The scene is built using a
`IfcGeom::Iterator<float>`
with following settings
```
    settings.set(IfcGeom::IteratorSettings::APPLY_DEFAULT_MATERIALS, true);
    settings.set(IfcGeom::IteratorSettings::CONVERT_BACK_UNITS, false);
    settings.set(IfcGeom::IteratorSettings::DISABLE_OPENING_SUBTRACTIONS, false);
    settings.set(IfcGeom::IteratorSettings::DISABLE_TRIANGULATION, false);
    settings.set(IfcGeom::IteratorSettings::EXCLUDE_SOLIDS_AND_SURFACES, false);
    settings.set(IfcGeom::IteratorSettings::FASTER_BOOLEANS, true);
    settings.set(IfcGeom::IteratorSettings::INCLUDE_CURVES, true);
    settings.set(IfcGeom::IteratorSettings::SEW_SHELLS, false);
    settings.set(IfcGeom::IteratorSettings::USE_BREP_DATA, false);
    settings.set(IfcGeom::IteratorSettings::USE_WORLD_COORDS, false);
```

**Comment by aothms** [comment]
Thanks for the report, I will have a look.

Kind regards,
Thomas

**Comment by aothms** [comment]
I cannot reproduce this. I have tested this with the OBJ serializer which directly emits the `IfcGeom::Iterator<float>` data. The amount of faces have stayed the same and all have positive area. On account of this I did commit a fix to prevent references to vertex normals being emitted as part of faces when --weld-vertices is set https://github.com/IfcOpenShell/IfcOpenShell/commit/9909e2f7e2b8da3f68ffffb0783d4700f6e796c5.

![weldvertices](https://cloud.githubusercontent.com/assets/1096535/15504781/e6b587e4-21c0-11e6-8647-46516a712496.png)
